From 4962c6e9fbdc542e8eee4b79714f4afd5da34b73 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 21 Dec 2021 22:18:04 +0000
Subject: [PATCH] gnu: cl-ana: Update to 0.0.0-2.e051002

* gnu/packages/lisp-xyz.scm:
  Apply new style for inputs.
  [native-inputs]: Add openmpi and pkg-config to pass build for hdf-cffi
  [propagated-inputs]: Add gnuplot
---
 gnu/packages/lisp-xyz.scm | 104 ++++++++++++++++++++------------------
 1 file changed, 56 insertions(+), 48 deletions(-)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index 407d44e156..94a562bb84 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -85,6 +85,7 @@ (define-module (gnu packages lisp-xyz)
   #:use-module (gnu packages lisp-check)
   #:use-module (gnu packages maths)
   #:use-module (gnu packages mp3)
+  #:use-module (gnu packages mpi)
   #:use-module (gnu packages ncurses)
   #:use-module (gnu packages networking)
   #:use-module (gnu packages pkg-config)
@@ -8103,57 +8104,64 @@ (define-public ecl-external-program
   (sbcl-package->ecl-package sbcl-external-program))
 
 (define-public sbcl-cl-ana
-  (let ((commit "fa7cee4c50aa1c859652813049ba0da7c18a0df9")
-        (revision "1"))
+  (let ((commit "e05100296e2c755a39f48e803627e3c0fd0c6e1d")
+        (revision "2")) ;; 20211221T204115+0000
     (package
-     (name "sbcl-cl-ana")
-     (version (git-version "0.0.0" revision commit))
-     (source
-      (origin
-       (method git-fetch)
-       (uri (git-reference
-             (url "https://github.com/ghollisjr/cl-ana")
-             (commit commit)))
-       (file-name (git-file-name name version))
-       (sha256
-        (base32 "0mr47l57m276dbpap7irr4fcnk5fgknhf6mgv4043s8h73amk5qh"))))
-     (build-system asdf-build-system/sbcl)
-     (native-inputs
-      (list sbcl-cl-fad))
-     (inputs
-      `(("alexandria" ,sbcl-alexandria)
-        ("antik" ,sbcl-antik)
-        ("cffi" ,sbcl-cffi)
-        ("cl-csv" ,sbcl-cl-csv)
-        ("closer-mop" ,sbcl-closer-mop)
-        ("external-program" ,sbcl-external-program)
-        ("gsl" ,gsl)
-        ("gsll" ,sbcl-gsll)
-        ("hdf5" ,hdf5-parallel-openmpi)
-        ("iterate" ,sbcl-iterate)
-        ("libffi" ,libffi)
-        ("split-sequence" ,sbcl-split-sequence)))
-     (arguments
-      `(#:phases
-        (modify-phases %standard-phases
-          (add-after 'unpack 'fix-paths
-            (lambda* (#:key inputs #:allow-other-keys)
-              (substitute* "hdf-cffi/hdf-cffi.lisp"
-                (("/usr/lib/i386-linux-gnu/hdf5/serial/libhdf5.so")
-                 (search-input-file inputs "/lib/libhdf5.so")))
-              (substitute* "gsl-cffi/gsl-cffi.lisp"
-                (("define-foreign-library gsl-cffi" all)
-                 (string-append all " (:unix "
-                                (assoc-ref inputs "gsl")
-                                "/lib/libgsl.so)")))
-              #t)))))
-     (synopsis "Common Lisp data analysis library")
-     (description
-      "CL-ANA is a data analysis library in Common Lisp providing tabular and
+      (name "sbcl-cl-ana")
+      (version (git-version "0.0.0" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/ghollisjr/cl-ana")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "00gcgc2fnny526k92c2snp8iwd6shyf3j1hbbbhr3lxa1300rq5x"))))
+      (build-system asdf-build-system/sbcl)
+      (native-inputs
+       (list openmpi ;; for hdf-cffi
+             pkg-config
+             sbcl-cl-fad))
+      (inputs
+       (list gsl
+             hdf5-parallel-openmpi
+             libffi
+             sbcl-antik
+             sbcl-cffi
+             sbcl-cl-csv
+             sbcl-closer-mop
+             sbcl-external-program
+             sbcl-gsll
+             sbcl-iterate
+             sbcl-alexandria
+             sbcl-split-sequence))
+      (propagated-inputs
+       (list gnuplot)) ;; for gnuplot-interface
+      (arguments
+       `(#:phases
+         (modify-phases %standard-phases
+           (add-after 'unpack 'fix-paths
+             (lambda* (#:key inputs #:allow-other-keys)
+               (substitute* "hdf-cffi/src/library.lisp"
+                 (("libhdf5.so")
+                  (search-input-file inputs "/lib/libhdf5.so")))
+               (substitute* "gsl-cffi/gsl-cffi.lisp"
+                 (("define-foreign-library gsl-cffi" all)
+                  ;; NOTE: (Sharlatan-20211221T215949+0000): Preposed to
+                  ;; upstream, modify package when it's merged.
+                  ;; https://github.com/ghollisjr/cl-ana/pull/44
+                  (string-append all " (:unix "
+                                 (assoc-ref inputs "gsl")
+                                 "/lib/libgsl.so)")))
+               #t)))))
+      (synopsis "Common Lisp data analysis library")
+      (description
+       "CL-ANA is a data analysis library in Common Lisp providing tabular and
 binned data analysis along with nonlinear least squares fitting and
 visualization.")
-     (home-page "https://github.com/ghollisjr/cl-ana")
-     (license license:gpl3))))
+      (home-page "https://github.com/ghollisjr/cl-ana")
+      (license license:gpl3))))
 
 (define-public cl-ana
   (sbcl-package->cl-source-package sbcl-cl-ana))
-- 
2.34.0

